datafusion/src/physical_optimizer/repartition.rs

```diff
@@ -219,20 +384,64 @@ mod tests {
             "ParquetExec: limit=None, partitions=[x]",
         ];
 
-        assert_eq!(&trim_plan_display(&plan), &expected);
+        assert_optimized!(expected, plan);
         Ok(())
     }
 
     #[test]
-    fn repartition_ignores_limit() -> Result<()> {
-        let optimizer = Repartition {};
+    fn repartition_unsorted_limit() -> Result<()> {
```

Comment:
new plans showing that data isn't repartitioned below limits if sorts are present